pkg/ignore/ignore.go

```diff
@@ -0,0 +1,127 @@
+package ignore
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+)
+
+type DockerIgnorer struct {}
+
+func (b *DockerIgnorer) Ignore(config *api.Config) error {
+	// so, to duplicate the .dockerignore capabilities (https://docs.docker.com/reference/builder/#dockerignore-file)
+	// we have a flow that follows:
+	/*
+
+		0) First note, .dockerignore rules are NOT recursive (unlike .gitignore) .. you have to list subdir explicitly
+		1) Read in the exclusion patterns
+		2) Skip over comments (noted by #)
+		3) note overrides (via exclamation sign i.e. !) and reinstate files (don't remove) as needed
+		4) leverage Glob matching to build list, as .dockerignore is documented as following filepath.Match / filepath.Glob
+		5) del files
+
+                 1 to 4 is in getListOfFilesToIgnore
+
+	*/
+
+	filesToDel, lerr := getListOfFilesToIgnore(config)
+	if lerr != nil {
+		return lerr
+	}
+
+	if filesToDel == nil {
+		return nil
+	}
+
+	// delete compiled list of files 
+	for _, fileToDel := range filesToDel {
+		glog.V(5).Infof("attempting to remove file %s \n", fileToDel)
+		rerr := os.RemoveAll(fileToDel)
+		if rerr != nil {
+			glog.Errorf("error removing file %s because of %v \n", fileToDel, rerr)
+			return rerr
+		}
+	}
+
+	return nil
+}
+
+func getListOfFilesToIgnore(config *api.Config) (map[string]string, error) {
+	path := filepath.Join(config.WorkingSourceDir, api.DefaultIgnoreFile)
+	file, err := os.Open(path)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			glog.Errorf("Ignore processing, problem opening %s because of %v\n", path, err)
+			return nil, err
+		}
+		glog.V(4).Info(".s2iignore file does not exist")
+		return nil, nil
+	}
+	defer file.Close()
+
+	filesToDel := make(map[string]string)
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		filespec := scanner.Text()
+		filespec = strings.Trim(filespec, " ")
+		glog.V(4).Infof(".s2iignore lists a file spec of %s \n", filespec)
```

Comment:
OK ... I was originally interested in confirming via trace that "#" were skipped, but yeah, at this point, that is confirmed by other means.  Part of next push.